File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        HashSet<Integer>set=new HashSet<>();
        for(int i=0;i<nums.length;i++){
            set.add(nums[i]);
        }
        int temp=nums.length/2;
        if(set.size()>temp)
            return temp;
        answer=set.size();
        
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0    
    #n/2마리를 선택하려고 하는데, 폰켓몬을 어떻게 고를 것이냐??
    temp=set(nums)
    if len(temp)>=len(nums)//2:
        return len(nums)//2
    else:
        return len(temp)
    
    
    return answer